sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -115,8 +119,8 @@ def plot(self, *, include_values=True, cmap='viridis',
             display_labels = np.arange(n_classes)
         else:
             display_labels = self.display_labels
-
-        fig.colorbar(self.im_, ax=ax)
+        if colorbar:
+            fig.colorbar(self.im_, ax=ax)
```

Comment:
@amueller Here, I have added if condition to make sure colorbar is added only if parameter is true and by default its value is true. Also, I have checked the code on a working model.